src/iterator.jl

```diff
@@ -48,14 +48,14 @@ macro implement_iteration(t)
         function Base.iterate(a::$t, i)
             datacur, bi, bstate = i
             (chunkiter, innerinds) = bstate
-            cistateold = chunkiter.taken
+            cistateold = length(chunkiter)
```

Comment:
I think this should rather be something like `total_length-length(chunkiter)`. The `.taken` field was counting how many items have already been consumed while `length` returns the number of remaining items.

Reply:
But its only used for comparing iterator state? I thought that would mean it's equivalent either way. And the tests pass...